### PR TITLE
Kafka RBAC changes for Pre-prod env

### PR DESCRIPTION
**Pull request by ponny-m**
## What was the problem?
At this time, Kafka is not able to connect and receive data in Pre-prod.

Associated tickets or Slack threads:
- #2916 

## How does this fix it?[^1]
Updating Kafka to use RBAC in Pre-prod environment for authentication and authorization will ensure we remain compliant. 

## How to test this PR
- Check pre-prod Kafka logs on Lens to confirm successful connection.

### CODE REVIEW

File: svc-bie-kafka/src/main/resources/application-prod-test.yaml [hunk 1]
```diff
@@ -1,35 +1,32 @@
...
-            type: PKCS12
-            location: "${TRUSTSTORE_FILE}"
-            password: "${BIE_KAFKA_TRUSTSTORE_PASSWORD}"
+          trust-store-location: "file:${TRUSTSTORE_FILE}"
+          trust-store-password: "${BIE_KAFKA_TRUSTSTORE_PASSWORD}"
+          trust-store-type: "PKCS12"
 
 bie:
-  kakfa-topic-prefix: "PRE_"
+  kakfa-topic-prefix: "PRE"
```
Comment (lisac) [inline svc-bie-kafka/src/main/resources/application-prod-test.yaml:32]: question: do we need to use a different key here, as part of the shift to RBAC? 
```suggestion
  kakfa-topic-infix: "PRE"
```

i noticed `kafka-topic-infix` is what is being using on QA and sandbox: 
https://github.com/department-of-veterans-affairs/abd-vro/blob/7938c9c2168f63eb4ff28d34dbfd04a96fcd0eea/svc-bie-kafka/src/main/resources/application-qa.yaml#L32
https://github.com/department-of-veterans-affairs/abd-vro/blob/7938c9c2168f63eb4ff28d34dbfd04a96fcd0eea/svc-bie-kafka/src/main/resources/application-sandbox.yaml#L32

Comment (chengjie8) [reply svc-bie-kafka/src/main/resources/application-prod-test.yaml:32]: Yes, the BIE team has updated all of their topic names, so the environment name has now been moved from the prefix to the middle of the topic names

File: svc-bie-kafka/src/main/resources/application-prod-test.yaml [hunk 2]
```diff
@@ -1,35 +1,32 @@
...
+      sasl:
+        mechanism: PLAIN
+        jaas:
+          config: "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"${BIE_KAFKA_RBAC_USERNAME}\" password=\"${BIE_KAFKA_RBAC_PASSWORD}\";"
     consumer:
-      group-id: "${BIE_KAFKA_PLACEHOLDERS_GROUP_ID:vro-bie-pre-vro}"
+      group-id: "${BIE_KAFKA_PLACEHOLDERS_GROUP_ID:EXT_VRO_PRE}"
       key-deserializer: "org.apache.kafka.common.serialization.StringDeserializer"
       value-deserializer: "io.confluent.kafka.serializers.KafkaAvroDeserializer"
       properties:
```
Comment (chengjie8) [inline svc-bie-kafka/src/main/resources/application-prod-test.yaml]: Should we make the security.protocol be in the same structure as the ones from dev and qa? Like this
```
    consumer:
      group-id: "${BIE_KAFKA_PLACEHOLDERS_GROUP_ID:EXT_VRO_PRE}"
      key-deserializer: "org.apache.kafka.common.serialization.StringDeserializer"
      value-deserializer: "io.confluent.kafka.serializers.KafkaAvroDeserializer"
      security:
        protocol: SASL_SSL
      ssl:
        trust-store-location: "file:${TRUSTSTORE_FILE}"
        trust-store-password: "${BIE_KAFKA_TRUSTSTORE_PASSWORD}"
        trust-store-type: "PKCS12"
```

Comment (ponny-m) [reply svc-bie-kafka/src/main/resources/application-prod-test.yaml]: The correct prefix if PRE and the boostrap and schemaregistry have preprod in the links.

**Review by tejans24 — APPROVED** [review]
LGTM, assuming all comments have already been addressed.

**Review by chengjie8 — APPROVED** [review]

**Pull request merged**